internal/ethapi/api.go

```diff
@@ -1017,14 +1032,42 @@ func (diff *BlockOverrides) Apply(blockCtx *vm.BlockContext) {
 	if diff.GasLimit != nil {
 		blockCtx.GasLimit = uint64(*diff.GasLimit)
 	}
-	if diff.Coinbase != nil {
-		blockCtx.Coinbase = *diff.Coinbase
+	if diff.FeeRecipient != nil {
+		blockCtx.Coinbase = *diff.FeeRecipient
+	}
+	if diff.PrevRandao != nil {
+		blockCtx.Random = diff.PrevRandao
+	}
+	if diff.BaseFeePerGas != nil {
+		blockCtx.BaseFee = diff.BaseFeePerGas.ToInt()
+	}
+}
+
+// ApplyToHeader overrides the given fields into a header.
+func (diff *BlockOverrides) ApplyToHeader(header *types.Header) {
```

Comment:
Sorry for the very late reply. I applied your suggestion. I opted for copying the header and applying the override fields. The copy is not really necessary since overrides are applied to a new header object, but as you mentioned it makes this method safer.